From b12c1e33939ac51d37f5f3c2cb1a31ce3f1e2415 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Sat, 15 Aug 2015 19:17:58 -0700
Subject: [PATCH] libmedia: Add 1440p camcorder quality

Add support for 2.5k 1440p camcorder quality (including timelapse).

[YumeMichi: Adjusted for Q]

Change-Id: Ib8dc7567ccae3f575185fad9385fbaf5830ebac0
---
 media/libmedia/MediaProfiles.cpp             | 2 ++
 media/libmedia/include/media/MediaProfiles.h | 6 ++++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/media/libmedia/MediaProfiles.cpp b/media/libmedia/MediaProfiles.cpp
index 98c5497..5d76e74 100644
--- a/media/libmedia/MediaProfiles.cpp
+++ b/media/libmedia/MediaProfiles.cpp
@@ -103,8 +103,10 @@ const MediaProfiles::NameToTagMap MediaProfiles::sCamcorderQualityNameMap[] = {
     {"highspeed4kdci", CAMCORDER_QUALITY_HIGH_SPEED_4KDCI},
     {"qhd", CAMCORDER_QUALITY_QHD},
     {"2k", CAMCORDER_QUALITY_2k},
+    {"1440p", CAMCORDER_QUALITY_1440P},
     {"timelapseqhd", CAMCORDER_QUALITY_TIME_LAPSE_QHD},
     {"timelapse2k", CAMCORDER_QUALITY_TIME_LAPSE_2k},
+    {"timelapse1440p", CAMCORDER_QUALITY_TIME_LAPSE_1440P},
 };
 
 #if LOG_NDEBUG
diff --git a/media/libmedia/include/media/MediaProfiles.h b/media/libmedia/include/media/MediaProfiles.h
index 3e8e7c8..22acfa0 100644
--- a/media/libmedia/include/media/MediaProfiles.h
+++ b/media/libmedia/include/media/MediaProfiles.h
@@ -38,7 +38,8 @@ enum camcorder_quality {
     CAMCORDER_QUALITY_4KDCI = 10,
     CAMCORDER_QUALITY_QHD = 11,
     CAMCORDER_QUALITY_2k = 12,
-    CAMCORDER_QUALITY_LIST_END = 12,
+    CAMCORDER_QUALITY_1440P = 13,
+    CAMCORDER_QUALITY_LIST_END = 13,
 
     CAMCORDER_QUALITY_TIME_LAPSE_LIST_START = 1000,
     CAMCORDER_QUALITY_TIME_LAPSE_LOW  = 1000,
@@ -54,7 +55,8 @@ enum camcorder_quality {
     CAMCORDER_QUALITY_TIME_LAPSE_4KDCI = 1010,
     CAMCORDER_QUALITY_TIME_LAPSE_QHD = 1011,
     CAMCORDER_QUALITY_TIME_LAPSE_2k = 1012,
-    CAMCORDER_QUALITY_TIME_LAPSE_LIST_END = 1012,
+    CAMCORDER_QUALITY_TIME_LAPSE_1440P = 1013,
+    CAMCORDER_QUALITY_TIME_LAPSE_LIST_END = 1013,
 
     CAMCORDER_QUALITY_HIGH_SPEED_LIST_START = 2000,
     CAMCORDER_QUALITY_HIGH_SPEED_LOW  = 2000,
-- 
2.7.4

